src/puppeteer_utils.js

```diff
@@ -143,30 +150,59 @@ const enqueueRequestsFromClickableElements = async (page, selector, purls, reque
  *   Puppeteer <a href="https://pptr.dev/#?product=Puppeteer&show=api-class-page" target="_blank"><code>Page</code></a> object.
  * @param {String} selector
  *   CSS selector matching elements to be clicked.
- * @param {PseudoUrl[]} pseudoUrls
- *   An array of {@link PseudoUrl}s matching the URLs to be enqueued.
  * @param {RequestQueue} requestQueue
- *   Request queue object where URLs will be enqueued.
+ *   {@link RequestQueue} instance where URLs will be enqueued.
+ * @param {PseudoUrl[]|Object[]|String[]} [pseudoUrls]
+ *   An array of {@link PseudoUrl}s matching the URLs to be enqueued,
+ *   or an array of Strings or Objects from which the {@link PseudoUrl}s should be constructed
+ *   The Objects must include at least a `purl` property, which holds a pseudoUrl string.
+ *   All remaining keys will be used as the `requestTemplate` argument of the {@link PseudoUrl} constructor.
  * @return {Promise<RequestOperationInfo[]>}
  *   Promise that resolves to an array of {@link RequestOperationInfo} objects.
  * @memberOf puppeteer
  */
-const enqueueLinks = async (page, selector, pseudoUrls, requestQueue) => {
+const enqueueLinks = async (page, selector, requestQueue, pseudoUrls = []) => {
+    // TODO: Remove after v1.0.0 gets released.
+    // Check for pseudoUrls as a third parameter.
+    if (Array.isArray(requestQueue)) {
+        log.warning('Argument "pseudoUrls" as the third parameter to enqueueLinks() is deprecated. '
+            + 'Use enqueueLinks(page, selector, requestQueue, pseudoUrls) instead. "pseudoUrls" are now optional.');
+        const tmp = requestQueue;
+        requestQueue = pseudoUrls;
+        pseudoUrls = tmp;
+    }
+
     checkParamOrThrow(page, 'page', 'Object');
     checkParamOrThrow(selector, 'selector', 'String');
-    checkParamOrThrow(pseudoUrls, 'purls', 'Array');
     checkParamPrototypeOrThrow(requestQueue, 'requestQueue', [RequestQueue, RequestQueueLocal], 'Apify.RequestQueue');
+    checkParamOrThrow(pseudoUrls, 'pseudoUrls', 'Array');
+
+    // Check and cache the pseudoUrls
+    let pseudoUrlInstances = enqueueLinksCache.get(pseudoUrls);
```

Comment:
Isn't it better to cache each PseudoUrl separately, rather than the entire array? Maybe the user is reusing same array for multiple calls, and adding elements to it. I know user can also updated same PseudoUrl instance, but I think it's not very likely.

Reply:
I'll see how it'd look like.